packages/core/crates/core-strategy/src/aggregating.rs

```diff
@@ -0,0 +1,543 @@
+    /// Default is 60 seconds.
+    #[serde_as(as = "DurationSeconds<u64>")]
+    pub aggregation_timeout: Duration,
+
+    /// Indicates whether to also perform ticket redemption after the ticket aggregation
+    /// in a channel was successful.
+    /// Default is `true`.
+    pub redeem_after_aggregation: bool,
+}
+
+impl Default for AggregatingStrategyConfig {
+    fn default() -> Self {
+        Self {
+            aggregation_threshold: 100,
+            aggregation_timeout: Duration::from_secs(60),
+            redeem_after_aggregation: true,
+        }
+    }
+}
+
+/// Represents a strategy that starts aggregating tickets in a certain
+/// channel, once the amount of acknowledged tickets in that channel goes
+/// above the given threshold.
+/// Optionally, the strategy can also redeem the aggregated ticket, if the aggregation
+/// was successful.
+pub struct AggregatingStrategy<Db: HoprCoreEthereumDbActions, T, U> {
+    db: Arc<RwLock<Db>>,
+    tx_sender: TransactionSender,
+    ticket_aggregator: Arc<Mutex<TicketAggregationActions<T, U>>>,
+    cfg: AggregatingStrategyConfig,
+}
+
+impl<Db: HoprCoreEthereumDbActions, T, U> Display for AggregatingStrategy<Db, T, U> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", Strategy::Aggregating(Default::default()))
+    }
+}
+
+impl<Db: HoprCoreEthereumDbActions, T, U> AggregatingStrategy<Db, T, U> {
+    pub fn new(
+        cfg: AggregatingStrategyConfig,
+        db: Arc<RwLock<Db>>,
+        tx_sender: TransactionSender,
+        ticket_aggregator: TicketAggregationActions<T, U>,
+    ) -> Self {
+        Self {
+            cfg,
+            db,
+            tx_sender,
+            ticket_aggregator: Arc::new(Mutex::new(ticket_aggregator)),
+        }
+    }
+}
+
+#[async_trait(? Send)]
+impl<Db: HoprCoreEthereumDbActions + 'static, T, U> SingularStrategy for AggregatingStrategy<Db, T, U> {
+    async fn on_acknowledged_ticket(&self, ack: &AcknowledgedTicket) -> crate::errors::Result<()> {
+        let channel_id = ack.ticket.channel_id;
+
+        let channel = match self.db.read().await.get_channel(&channel_id).await? {
+            Some(channel) => channel,
+            None => {
+                error!("{self} strategy: encountered {ack} in a non-existing channel!");
+                return Err(ChannelDoesNotExist.into());
+            }
+        };
+
+        let acks_in_channel = self
+            .db
+            .read()
+            .await
+            .get_acknowledged_tickets(Some(channel.clone()))
+            .await?
+            .iter()
+            .filter(|ack| ack.status == AcknowledgedTicketStatus::Untouched)
+            .count() as u32;
+
+        if acks_in_channel < self.cfg.aggregation_threshold {
+            debug!(
+                "{self} strategy: {channel} has {acks_in_channel} < {} ack tickets, not aggregating yet",
+                self.cfg.aggregation_threshold
+            );
+            return Ok(());
+        }
+
```

Comment:
As an additional check, I'd look if there are any tickets already `BeingAggregated` in this channel, and return early to skip starting another aggregation.